File: client/src/models/common/Limit.ts
import {Option} from "../../components/common/InputSelectField";

export default interface Limit {
    type: LimitType
    limit: number;
}

export enum LimitType {
    None = 'None',
    PerRound = 'Per Round',
    PerEncounter = 'Per Encounter',
    PerSession = 'Per Session'
}

export const getLimitOptions = (): Option[] => {
    return Object.values(LimitType).map((value) => ({value}))
}

export class DefaultLimit {
    static create():Limit {
        return {
            type: LimitType.None,
            limit: 0
        }
    }
}

File: server/src/main/java/com/github/genraven/genesys/domain/Limit.java
package com.github.genraven.genesys.domain;

import com.fasterxml.jackson.annotation.JsonValue;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;

@Data
public class Limit {
    private Type type = Type.NONE;
    private int limit = 0;

    @AllArgsConstructor
    @Getter
    public enum Type {
        NONE("None"),
        PER_ROUND("Per Round"),
        PER_ENCOUNTER("Per Encounter"),
        PER_SESSION("Per Session");

        @JsonValue
        private final String label;
    }
}
